.github/workflows/main.yml

```
@@ -1,12 +1,16 @@
 name: build
 
-on: [push, pull_request]
+on:
+  push: # Run on pushes to the default branch
+    branches: [main,add-py-3_11-3_12-to-ci-test]
```

```suggestion
    branches: [main]
```
remove before merging